src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -86,6 +86,18 @@ public static WellKnownTypeProvider GetOrCreate(Compilation compilation)
         public bool TryGetOrCreateTypeByMetadataName(
             string fullTypeName,
             [NotNullWhen(returnValue: true)] out INamedTypeSymbol? namedTypeSymbol)
+        {
+            if (_fullNameToTypeMap.TryGetValue(fullTypeName, out namedTypeSymbol))
+            {
+                return namedTypeSymbol is not null;
+            }
+
+            return TryGetOrCreateTypeByMetadataNameSlow(fullTypeName, out namedTypeSymbol);
```

Comment:
Comment here would be nice, since the new method name doesn't really capture the intent of the change

Reply:
Added `[PerformanceSensitive]`